[libuuid] libuuid/1.0.3: does not build on AIX

Libuuid checks the Autoconf test macro HAVE_SYS_SYSCALL_H in gen_uuid.c before  including <sys/syscall.h>, but then doesn't  check when the same  header is included in randutils.c.   Because AIX  does   not provide this header,  the compilation fails abruptly.

I have submitted a PR to the upstream project as https://sourceforge.net/p/libuuid/code/merge-requests/2/

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libuuid/1.0.3**
  * Operating System+version: **AIX Power 7.1**
  * Compiler+version: **GCC 8**
  * Conan version: **conan 1.31.2**
  * Python version: **Python 3.8.4**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=AIX
os_build=AIX
os.version=7.1
compiler=gcc
compiler.version=8.4
compiler.libcxx=libstdc++
build_type=Release
arch=ppc64
arch_build=ppc64
[options]
[build_requires]
[env]
OBJECT_MODE=64
CFLAGS=-pthread -mcpu=power4 -maix64 -mvsx
CXXFLAGS=-pthread -mcpu=power4 -maix64 -mvsx
CONAN_CPU_COUNT=4
CC=/opt/freeware/bin/gcc
CXX=/opt/freeware/bin/g++
```


### Steps to reproduce (Include if Applicable)
Attempt to create a libuuid/1.0.3 package on AIX:

`conan create -pr gcc-8-aix71-64 --build=missing conanfile.py libuuid/1.0.3@_/_`


### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
(python-env-nepal) robb@nepal:/raid/proj/procyon/checkouts-procyon/robb/apdfl-18box/conan-center-index/recipes/libuuid/all (master)$ conan create -pr gcc-8-aix71-64 --build=missing conanfile.py libuuid/1.0.3@_/_
[HOOK - conan-center.py] pre_export(): [DEPRECATED GLOBAL CPPSTD (KB-H001)] OK
[HOOK - conan-center.py] pre_export(): [REFERENCE LOWERCASE (KB-H002)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE METADATA (KB-H003)] OK
[HOOK - conan-center.py] pre_export(): [HEADER_ONLY, NO COPY SOURCE (KB-H005)] OK
[HOOK - conan-center.py] pre_export(): [FPIC OPTION (KB-H006)] OK
[HOOK - conan-center.py] pre_export(): [VERSION RANGES (KB-H008)] OK
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] Total recipe size: 4.0224609375 KB
[HOOK - conan-center.py] pre_export(): [RECIPE FOLDER SIZE (KB-H009)] OK
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] exports: None
[HOOK - conan-center.py] pre_export(): [EXPORT LICENSE (KB-H023)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE FOLDER (KB-H024)] OK
[HOOK - conan-center.py] pre_export(): [META LINES (KB-H025)] OK
[HOOK - conan-center.py] pre_export(): [CONAN CENTER INDEX URL (KB-H027)] OK
[HOOK - conan-center.py] pre_export(): [CMAKE MINIMUM VERSION (KB-H028)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - RUN ENVIRONMENT (KB-H029)] OK
[HOOK - conan-center.py] pre_export(): [CONANDATA.YML FORMAT (KB-H030)] OK
[HOOK - conan-center.py] pre_export(): [TEST PACKAGE - NO IMPORTS() (KB-H034)] OK
[HOOK - conan-center.py] pre_export(): [NO AUTHOR (KB-H037)] OK
Exporting package recipe
libuuid/1.0.3 exports: File 'conandata.yml' found. Exporting it...
libuuid/1.0.3 exports: Copied 1 '.yml' file: conandata.yml
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'1.0.3': {'url': 'https://downloads.sourceforge.net/project/libuuid/libuuid-1.0.3.tar.gz', 'sha256': '46af3275291091009ad7f1b899de3d0cea0252737550e7919d17237997db5644'}}}
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] New conandata.yml contents: sources:
  1.0.3:
    sha256: 46af3275291091009ad7f1b899de3d0cea0252737550e7919d17237997db5644
    url: https://downloads.sourceforge.net/project/libuuid/libuuid-1.0.3.tar.gz

[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] OK
libuuid/1.0.3: A new conanfile.py version was exported
libuuid/1.0.3: Folder: /home/robb/.conan/data/libuuid/1.0.3/_/_/export
libuuid/1.0.3: Using the exported files summary hash as the recipe revision: 0af468907873ba6cfa5cd2a28966a8af 
libuuid/1.0.3: Exported revision: 0af468907873ba6cfa5cd2a28966a8af
Configuration:
[settings]
arch=ppc64
arch_build=ppc64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=8.4
os=AIX
os.version=7.1
os_build=AIX
[options]
[build_requires]
[env]
CC=/opt/freeware/bin/gcc
CFLAGS=-pthread -mcpu=power4 -maix64 -mvsx
CONAN_CPU_COUNT=4
CXX=/opt/freeware/bin/g++
CXXFLAGS=-pthread -mcpu=power4 -maix64 -mvsx
OBJECT_MODE=64
b2:NO_CXX11_CHECK=true
icu:PATH=[/opt/freeware/bin]
libuuid/1.0.3 (test package): Installing package
Requirements
    libuuid/1.0.3 from local cache - Cache
Packages
    libuuid/1.0.3:0150ece77320347941f57bde2644131f805e4bbe - Build

Installing (downloading, building) binaries...
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
libuuid/1.0.3: Configuring sources in /home/robb/.conan/data/libuuid/1.0.3/_/_/source
Downloading libuuid-1.0.3.tar.gz completed [310.80k]                                     

[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
libuuid/1.0.3: Copying sources to build folder
libuuid/1.0.3: Building your package in /home/robb/.conan/data/libuuid/1.0.3/_/_/build/0150ece77320347941f57bde2644131f805e4bbe
libuuid/1.0.3: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
libuuid/1.0.3: Calling build()
libuuid/1.0.3: Calling:
 > ./configure '--enable-shared=no' '--enable-static=yes' '--prefix=/home/robb/.conan/data/libuuid/1.0.3/_/_/package/0150ece77320347941f57bde2644131f805e4bbe' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share' 
checking for a BSD-compatible install... /opt/freeware/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /opt/freeware/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... powerpc-ibm-aix7.1.0.0
checking host system type... powerpc-ibm-aix7.1.0.0
checking how to print strings... print -r
checking for style of include used by make... GNU
checking for gcc... /opt/freeware/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /opt/freeware/bin/gcc accepts -g... no
checking for /opt/freeware/bin/gcc option to accept ISO C89... none needed
checking dependency style of /opt/freeware/bin/gcc... gcc3
checking for a sed that does not truncate output... /opt/freeware/bin/sed
checking for grep that handles long lines and -e... /opt/freeware/bin/grep
checking for egrep... /opt/freeware/bin/grep -E
checking for fgrep... /opt/freeware/bin/grep -F
checking for ld used by /opt/freeware/bin/gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 786432
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... no
checking how to convert powerpc-ibm-aix7.1.0.0 file names to powerpc-ibm-aix7.1.0.0 format... func_convert_file_noop
checking how to convert powerpc-ibm-aix7.1.0.0 file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... print -r --
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /opt/freeware/bin/gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... /opt/freeware/bin/gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /opt/freeware/bin/gcc supports -fno-rtti -fno-exceptions... no
checking for /opt/freeware/bin/gcc option to produce PIC...  -DPIC
checking if /opt/freeware/bin/gcc PIC flag  -DPIC works... yes
checking if /opt/freeware/bin/gcc static flag -static works... no
checking if /opt/freeware/bin/gcc supports -c -o file.o... yes
checking if /opt/freeware/bin/gcc supports -c -o file.o... (cached) yes
checking whether the /opt/freeware/bin/gcc linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... aix7.1.0.0 ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... no
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for gcc... (cached) /opt/freeware/bin/gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether /opt/freeware/bin/gcc accepts -g... (cached) no
checking for /opt/freeware/bin/gcc option to accept ISO C89... (cached) none needed
checking dependency style of /opt/freeware/bin/gcc... (cached) gcc3
checking for library containing socket... none required
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking for inttypes.h... (cached) yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/file.h usability... yes
checking sys/file.h presence... yes
checking for sys/file.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking whether _SC_HOST_NAME_MAX is declared... yes
checking for int32_t... yes
checking for mode_t... yes
checking for size_t... yes
checking for ssize_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for uint8_t... yes
checking for ftruncate... yes
checking for gettimeofday... yes
checking for memset... yes
checking for socket... yes
checking for strtoul... yes
checking for usleep... yes
checking for srandom... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating uuid.pc
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
        make  all-am
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-clear.lo -MD -MP -MF .deps/libuuid_la-clear.Tpo -c -o libuuid_la-clear.lo `test -f 'clear.c' || echo './'`clear.c
        mv -f .deps/libuuid_la-clear.Tpo .deps/libuuid_la-clear.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-compare.lo -MD -MP -MF .deps/libuuid_la-compare.Tpo -c -o libuuid_la-compare.lo `test -f 'compare.c' || echo './'`compare.c
        mv -f .deps/libuuid_la-compare.Tpo .deps/libuuid_la-compare.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-copy.lo -MD -MP -MF .deps/libuuid_la-copy.Tpo -c -o libuuid_la-copy.lo `test -f 'copy.c' || echo './'`copy.c
        mv -f .deps/libuuid_la-copy.Tpo .deps/libuuid_la-copy.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-gen_uuid.lo -MD -MP -MF .deps/libuuid_la-gen_uuid.Tpo -c -o libuuid_la-gen_uuid.lo `test -f 'gen_uuid.c' || echo './'`gen_uuid.c
        mv -f .deps/libuuid_la-gen_uuid.Tpo .deps/libuuid_la-gen_uuid.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-isnull.lo -MD -MP -MF .deps/libuuid_la-isnull.Tpo -c -o libuuid_la-isnull.lo `test -f 'isnull.c' || echo './'`isnull.c
        mv -f .deps/libuuid_la-isnull.Tpo .deps/libuuid_la-isnull.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-pack.lo -MD -MP -MF .deps/libuuid_la-pack.Tpo -c -o libuuid_la-pack.lo `test -f 'pack.c' || echo './'`pack.c
        mv -f .deps/libuuid_la-pack.Tpo .deps/libuuid_la-pack.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-parse.lo -MD -MP -MF .deps/libuuid_la-parse.Tpo -c -o libuuid_la-parse.lo `test -f 'parse.c' || echo './'`parse.c
        mv -f .deps/libuuid_la-parse.Tpo .deps/libuuid_la-parse.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-unpack.lo -MD -MP -MF .deps/libuuid_la-unpack.Tpo -c -o libuuid_la-unpack.lo `test -f 'unpack.c' || echo './'`unpack.c
        mv -f .deps/libuuid_la-unpack.Tpo .deps/libuuid_la-unpack.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-unparse.lo -MD -MP -MF .deps/libuuid_la-unparse.Tpo -c -o libuuid_la-unparse.lo `test -f 'unparse.c' || echo './'`unparse.c
        mv -f .deps/libuuid_la-unparse.Tpo .deps/libuuid_la-unparse.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-uuid_time.lo -MD -MP -MF .deps/libuuid_la-uuid_time.Tpo -c -o libuuid_la-uuid_time.lo `test -f 'uuid_time.c' || echo './'`uuid_time.c
        mv -f .deps/libuuid_la-uuid_time.Tpo .deps/libuuid_la-uuid_time.Plo
        /bin/sh ./libtool  --tag=CC   --mode=compile /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I.   -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-randutils.lo -MD -MP -MF .deps/libuuid_la-randutils.Tpo -c -o libuuid_la-randutils.lo `test -f 'randutils.c' || echo './'`randutils.c
        mv -f .deps/libuuid_la-randutils.Tpo .deps/libuuid_la-randutils.Plo
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-copy.lo -MD -MP -MF .deps/libuuid_la-copy.Tpo -c copy.c -o libuuid_la-copy.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-compare.lo -MD -MP -MF .deps/libuuid_la-compare.Tpo -c compare.c -o libuuid_la-compare.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-unpack.lo -MD -MP -MF .deps/libuuid_la-unpack.Tpo -c unpack.c -o libuuid_la-unpack.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-uuid_time.lo -MD -MP -MF .deps/libuuid_la-uuid_time.Tpo -c uuid_time.c -o libuuid_la-uuid_time.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-isnull.lo -MD -MP -MF .deps/libuuid_la-isnull.Tpo -c isnull.c -o libuuid_la-isnull.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-parse.lo -MD -MP -MF .deps/libuuid_la-parse.Tpo -c parse.c -o libuuid_la-parse.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-clear.lo -MD -MP -MF .deps/libuuid_la-clear.Tpo -c clear.c -o libuuid_la-clear.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-gen_uuid.lo -MD -MP -MF .deps/libuuid_la-gen_uuid.Tpo -c gen_uuid.c -o libuuid_la-gen_uuid.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-pack.lo -MD -MP -MF .deps/libuuid_la-pack.Tpo -c pack.c -o libuuid_la-pack.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-unparse.lo -MD -MP -MF .deps/libuuid_la-unparse.Tpo -c unparse.c -o libuuid_la-unparse.o
libtool: compile:  /opt/freeware/bin/gcc -DHAVE_CONFIG_H -I. -DNDEBUG -I. -maix64 -O3 -s -fPIC -pthread -mcpu=power4 -maix64 -mvsx -MT libuuid_la-randutils.lo -MD -MP -MF .deps/libuuid_la-randutils.Tpo -c randutils.c -o libuuid_la-randutils.o
randutils.c:16:10: fatal error: sys/syscall.h: No such file or directory
 #include <sys/syscall.h>
          ^~~~~~~~~~~~~~~
compilation terminated.
make: 1254-004 The error code from the last command is 1.


Stop.
make: 1254-004 The error code from the last command is 2.


Stop.
libuuid/1.0.3: 
libuuid/1.0.3: ERROR: Package '0150ece77320347941f57bde2644131f805e4bbe' build failed
libuuid/1.0.3: WARN: Build folder /home/robb/.conan/data/libuuid/1.0.3/_/_/build/0150ece77320347941f57bde2644131f805e4bbe
ERROR: libuuid/1.0.3: Error in build() method, line 48
        autotools.make()
        ConanException: Error 2 while executing make -j4
(python-env-nepal) robb@nepal:/raid/proj/procyon/checkouts-procyon/robb/apdfl-18box/conan-center-index/recipes/libuuid/all (master)$ 
```

</details>

--- a/recipes/libuuid/all/conandata.yml
+++ b/recipes/libuuid/all/conandata.yml
@@ -2,3 +2,7 @@ sources:
   "1.0.3":
     url: "https://downloads.sourceforge.net/project/libuuid/libuuid-1.0.3.tar.gz"
     sha256: "46af3275291091009ad7f1b899de3d0cea0252737550e7919d17237997db5644"
+patches:
+  "1.0.3":
+    - patch_file: "patches/0001-Check-for-HAVE_SYS_SYSCALL_H.patch"
+      base_path: "source_subfolder"

--- a/recipes/libuuid/all/conanfile.py
+++ b/recipes/libuuid/all/conanfile.py
@@ -11,6 +11,7 @@ class LibuuidConan(ConanFile):
     license = "BSD-3-Clause"
     topics = ("conan", "libuuid", "uuid", "unique-id", "unique-identifier")
     settings = "os", "arch", "compiler", "build_type"
+    exports_sources = "patches/**"
     options = {"shared": [True, False], "fPIC": [True, False]}
     default_options = {"shared": False, "fPIC": True}
     _source_subfolder = "source_subfolder"
@@ -19,6 +20,10 @@ class LibuuidConan(ConanFile):
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
 
     def config_options(self):
         if self.settings.os == 'Windows':
@@ -43,6 +48,7 @@ class LibuuidConan(ConanFile):
         return self._autotools
 
     def build(self):
+        self._patch_sources()
         with tools.chdir(self._source_subfolder):
             autotools = self._configure_autotools()
             autotools.make()

--- a/recipes/libuuid/all/patches/0001-Check-for-HAVE_SYS_SYSCALL_H.patch
+++ b/recipes/libuuid/all/patches/0001-Check-for-HAVE_SYS_SYSCALL_H.patch
@@ -1,0 +1,32 @@
+From 32dba1f82848e2bf4e2546dac06296fa7b387dff Mon Sep 17 00:00:00 2001
+From: Rob Boehne <robb@datalogics.com>
+Date: Mon, 16 Nov 2020 09:06:42 -0600
+Subject: [PATCH] Check for HAVE_SYS_SYSCALL_H
+
+Fix a compilation error on systems without sys/syscall.h by applying the same checks used elsewhere.
+---
+ randutils.c | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/randutils.c b/randutils.c
+index 80893d3..8bf3af6 100644
+--- a/randutils.c
++++ b/randutils.c
+@@ -11,9 +11,14 @@
+ #include <fcntl.h>
+ #include <stdlib.h>
+ #include <string.h>
++
++#if defined(HAVE_SYS_TIME_H)
+ #include <sys/time.h>
++#endif
+ 
++#if defined(__linux__) && defined(HAVE_SYS_SYSCALL_H)
+ #include <sys/syscall.h>
++#endif
+ 
+ #include "randutils.h"
+ 
+-- 
+2.15.2
+

--- a/recipes/libuuid/all/test_package/CMakeLists.txt
+++ b/recipes/libuuid/all/test_package/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.11)
+cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)

